specs/number-field.md

```diff
@@ -0,0 +1,123 @@
+# Number Field
+
+## Overview
+
+An implementation of a generic input for numerical values as a form-connected web-component.
+
+### Use Cases
+
+Used anywhere an author might otherwise use input[type="number"].
+
+### Features
+
+1. Form association
+
+2. Conversion to/from number/string between the view and the control
+
+### Prior Art/Examples
+
+- [FAST-DNA (React)](https://explore.fast.design/components/number-field)
+- [Carbon Design](https://www.carbondesignsystem.com/components/number-input/code)
+- [Ant Design](https://ant.design/components/input-number/)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/spinbutton)
+- [Duet](https://www.duetds.com/components/number-input/)
+
+---
+
+### API
+
+Extends Form associated component
+
+*Component Name*
+- `fast-number-field`
+
+*Attrs* - Adapted from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number)
+- `autofocus` - automatically focuses the control
+- `readonly` - the number field should be submitted with the form but should not be editable.
+- `placeholder` - an exemplar value to display in the input field whenever it is empty
+- `required` - boolean value that sets the field as required
+- `value` - string value of the number field, can be an empty string
+- `disabled` - disables the control
+- `list` - the id of the `<datalist>` element that contains the optional pre-defined autocomplete options
+- `max`	- the maximum value to accept for this input
+- `min` -	the minimum number to accept for this input
+- `maxlength`	- the maximum number of characters the input should accept
+- `minlength` -	the minimum number of characters long the input can be and still be considered valid
+- `name` - the name of the control
+- `size` - a number indicating how many characters wide the input field should be
+- `appearance` - enum
+  - outline (default)
+  - filled
```

Comment:
Should we also include whether or not the number can contain a decimal and how many decimal places it supports?

Reply:
Should we support a `step` property that indicates how much to increment/decrement by? As an example, imagine an audio app that wants to represent tuning in KHz. They might want to have an increment of .1 KHz.

Reply:
I think step for sure should be added, I'll update with that.

Reply:
added step 👍